airflow/models/dagbag.py

```diff
@@ -688,10 +688,10 @@ def _sync_perm_for_dag(cls, dag: DAG, session: Session = NEW_SESSION):
 
         root_dag_id = dag.parent_dag.dag_id if dag.parent_dag else dag.dag_id
 
-        def needs_perms(dag_id: str) -> bool:
+        def has_perms(dag_id: str) -> bool:
```

Comment:
It's more efficient to see if you need perms, as you can short circuit out on the first miss.

This, as written, is wrong because you don't check that a perm exists for everything in DAG_ACTIONS.